### DISCUSSION TITLE
Rename CLV Models for Specific Use Case?

### DISCUSSION THREAD

**Discussion by ricardoV94** [pymc-labs/pymc-marketing#1229] [Ideas]
That type of description seems to exist for every model we are implementing in the CLV, I think it makes the models much more "discoverable" than their weird names. We should include that!

![image](https://user-images.githubusercontent.com/28983449/204761959-8d0e30bc-d87b-4693-a74a-72bd43d1cb85.png)

**Comment by ColtAllen** [comment]
While working on [btyd](https://github.com/ColtAllen/btyd), I got a request to add a chart to the documentation highlighting the differences between the various models.

On the matter of model discovery, I've been wondering if CLV is the most appropriate name for this module? The only model that deals with valuation in any way is the Gamma-Gamma model. All other models estimate if a customer is still active and future transaction behavior. "Buy till you die" is the most common moniker I've seen for non-contractual modeling.

**Comment by ricardoV94** [comment]
> While working on [btyd](https://github.com/ColtAllen/btyd), I got a request to add a chart to the documentation highlighting the differences between the various models.

This would be great as a documentation piece! Have you started working on this / do you have any ideas?

> On the matter of model discovery, I've been wondering if CLV is the most appropriate name for this module? The only model that deals with valuation in any way is the Gamma-Gamma model. All other models estimate if a customer is still active and future transaction behavior. "Buy till you die" is the most common moniker I've seen for non-contractual modeling.

I like the name of CLV as the sub-package. The `L` stands for the lifetime and the `V` for value, no?

**Comment by ColtAllen** [comment]
> I like the name of CLV as the sub-package. The `L` stands for the lifetime and the `V` for value, no?

After seeing that fake Bible verse about CLV posted on LinkedIn by Fader himself, I no longer have an issue with the name 🤣

**Comment by ColtAllen** [comment]
> This would be great as a documentation piece! Have you started working on this / do you have any ideas?

I'm still learning the nuances of these models (and there are many more yet to be added) but a key assumption of the BG/NBD model is that it treats all non-repeat customers as alive with probability 1, which may not be valid depending on the use case. The Modified BG/NBD model does not have this assumption, but seems to require more data for convergence.

The research papers for most of these models also contain two versions - modeling an individual customer, and modeling the average characteristics of the population. For most use cases, the latter is more realistic because many companies have millions of customers. However, for use cases involving oligopsony, the individual approach would be ideal.

Most models also requires input data in the form of Recency/Frequency/T, but I'm aware of two that require raw transactions - the Pareto/GGG and the the BG/NBD with time-varying covariates. I'm also of the opinion raw transactions are the best way to represent individual customers.

Speaking of covariates, I've read research papers containing implementations of the BG/NBD and Pareto/NBD models for time-variant and time-invariant covariates.

**Comment by juanitorduz** [comment]
Here is a concrete TODO:

- [x] BG/NBD https://github.com/pymc-labs/pymc-marketing/pull/571
- [x] Gamma-Gamma https://github.com/pymc-labs/pymc-marketing/pull/573
- [x] Pareto

**Comment by ColtAllen** [comment]
Renamed title from "Update Model Story. . ." to clarify core discussion. This is a problem that has come up numerous times in posted discussions and issues, and may be seriously inhibiting our promotional efforts and user awareness of these models.

In a discussion earlier this year, a user didn't realize `BetaGeoModel` and `ParetoNBDModel` are for the same use case, and another even tried to use them for subscriptions, which is the domain of `ShiftedBetaGeoModelIndividual`. However, no promotional or educational resources for SBG exist at all beyond `pymc-marketing` and the original research paper.

I've considered the following renaming conventions:

-  `ShiftedBetaGeoModelIndividual` -> SubscriptionModelIndividual
- `BetaGeoModel` -> RetailBetaGeoModel
- `ParetoNBDModel` -> RetailParetoNBDModel
- `GammaGammaModel` -> RetailSpendModel
- `BetaGeoBetaBinomModel` -> EventsModel
- `ExponentialGammaModel` -> ConversionTimeModel (this one hasn't been added yet)

An important caveat is that some of these have rather creative use cases: The `lifetimes` README mentions hospital patients as a "retail" example, and the "conversion" model can honestly be used for anything involving time-until-event in continuous time. In one of my more insane episodes I even requested my data from Hinge and used the "Subscription" model to predict the probability of another date (it didn't converge until I increased the NUTS target acceptance rate, hilariously enough).

Renaming these would obviously entail massive API-breaking changes, so best do so sooner rather than later. What are everyone's thoughts?
